my-app/src/app/modules/courses-page/courses-page.component.ts

```diff
@@ -16,18 +13,28 @@ import {FilterPipe} from "./core";
   providers: [FilterPipe]
 })
 export class CoursesPageComponent implements OnInit {
-  public courses: Course[] | undefined;
+  public initialCourses: Course[] | undefined;
+  public currentCourses: Course[] | undefined;
```

Comment:
Try to use empty array as an initial value

Reply:
Fixed